### PR TITLE
Refactored the node for easier code reuse.

### PR DESCRIPTION
**Pull request by SirVer**

### CODE REVIEW

File: cartographer_ros/src/sensor_data.cc [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ */
+
+#include "sensor_data.h"
+
+#include <string>
+
+#include "glog/logging.h"
+
+namespace cartographer_ros {
+namespace {
```
Comment (damonkohler) [inline cartographer_ros/src/sensor_data.cc]: Add blank line above.

File: cartographer_ros/src/sensor_data_producer.h [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+#include "sensor_data.h"
+#include "sensor_msgs/Imu.h"
+#include "sensor_msgs/LaserScan.h"
+#include "sensor_msgs/MultiEchoLaserScan.h"
+#include "sensor_msgs/PointCloud2.h"
+
+namespace cartographer_ros {
+
+// A wraper around SensorCollator that converts ROS messages into our internal
+// representation and passes it on to the 'sensor_collator'.
```
Comment (damonkohler) [inline cartographer_ros/src/sensor_data_producer.h]: passes _them_ on

File: cartographer_ros/src/sensor_data_producer.h [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+class SensorDataProducer {
+ public:
+  explicit SensorDataProducer(
+      const int trajectory_id,
```
Comment (damonkohler) [inline cartographer_ros/src/sensor_data_producer.h]: Remove const from prototype

File: cartographer_ros/src/sensor_data_producer.h [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+#include "nav_msgs/Odometry.h"
+#include "sensor_data.h"
+#include "sensor_msgs/Imu.h"
+#include "sensor_msgs/LaserScan.h"
+#include "sensor_msgs/MultiEchoLaserScan.h"
+#include "sensor_msgs/PointCloud2.h"
+
+namespace cartographer_ros {
+
+// A wraper around SensorCollator that converts ROS messages into our internal
```
Comment (wohe) [inline cartographer_ros/src/sensor_data_producer.h]: typo: wrapper

**Comment by damonkohler** [comment]
LGTM

**Pull request merged**